packages/internal/src/__tests__/graphqlCodeGen.test.ts

```diff
@@ -0,0 +1,218 @@
+  __typename?: 'Redwood';
+  currentUser?: Maybe<Scalars['JSON']>;
+  prismaVersion?: Maybe<Scalars['String']>;
+  version?: Maybe<Scalars['String']>;
+};`)
+  expect(gqlTypesApiOutput).toContain(`export type Todo = {
+  __typename?: 'Todo';
+  body: Scalars['String'];
+  id: Scalars['Int'];
+  status: Scalars['String'];
+};`)
+  expect(gqlTypesApiOutput).toContain('JSON?: GraphQLScalarType;')
+  expect(gqlTypesApiOutput)
+    .toContain(`export interface BigIntScalarConfig extends GraphQLScalarTypeConfig<ResolversTypes['BigInt'], any> {
+  name: 'BigInt';
+}`)
+})
+
+test('respects user provided codegen config', async () => {
+  const customCodegenConfigPath = path.join(FIXTURE_PATH, 'codegen.yml')
+
+  // Add codegen.yml to fixture folder
+  fs.writeFileSync(
+    customCodegenConfigPath,
+    `config:
+  omitOperationSuffix: false
+  namingConvention:
+    typeNames: change-case-all#upperCase`
+  )
+
+  // Wrapping in `try` to make sure codegen.yml is always deleted, even if the
+  // test fails
+  try {
+    await generateGraphQLSchema()
+    const [outputPath] = await generateTypeDefGraphQLWeb()
+
+    const gqlTypesOutput = fs.readFileSync(outputPath, 'utf-8')
+
+    // Should be upper cased type
+    expect(gqlTypesOutput).toContain('ADDTODO_CREATETODOMUTATION')
+
+    // because we override omitOperationSuffix to false, it should append QUERY
+    // for __fixtures__/example-todo-main/../NumTodosCell.js
+    expect(gqlTypesOutput).toContain('NUMTODOSCELL_GETCOUNTQUERY')
+  } finally {
+    // Delete added codegen.yml
+    fs.rmSync(customCodegenConfigPath)
+  }
+})
+
+test("Doesn't throw or print any errors with empty project", async () => {
+  const fixturePath = path.resolve(
+    __dirname,
+    '../../../../__fixtures__/empty-project'
+  )
+  process.env.RWJS_CWD = fixturePath
+  const oldConsoleError = console.error
+  console.error = jest.fn()
+
+  try {
+    await generateGraphQLSchema()
+    await generateTypeDefGraphQLWeb()
+    await generateTypeDefGraphQLApi()
+  } catch (e) {
+    console.error(e)
+    // Fail if any of the three above calls throws an error
+    expect(false).toBeTruthy()
+  }
+
+  try {
+    expect(console.error).not.toHaveBeenCalled()
+  } finally {
+    console.error = oldConsoleError
+    delete process.env.RWJS_CWD
+  }
+})
+
+test("Doesn't swallow legit errors - invalidQueryType", async () => {
+  const fixturePath = path.resolve(
+    __dirname,
+    './fixtures/typeDefinitions/invalidQueryType'
+  )
+  process.env.RWJS_CWD = fixturePath
+  const oldConsoleError = console.error
+  console.error = jest.fn()
+
+  await generateTypeDefGraphQLWeb({ logErrors: true })
+
+  try {
+    expect(console.error).toHaveBeenCalledWith(
+      expect.objectContaining({
+        message: expect.stringMatching(/field.*softKitten.*Query/),
+      })
+    )
+  } finally {
+    console.error = oldConsoleError
+    delete process.env.RWJS_CWD
+  }
+})
+
+test("Doesn't swallow legit errors - missingType", async () => {
```

Comment:
Do you think we could:
a) Move this into a separate describe block
b) Before **each**, delete all generated .d.ts files, to make sure the old ts files in the fixture aren't picked up
c) Make sure the generate .d.ts isn't commited in the missingType fixture

Reply:
a) done
b) I've (manually) deleted all .d.ts files. No "old" files exist
c) No files are generated as part of the tests, because the generation bails on the (expected) errors. If, for some reason, files starts being generated hopefully that's a red flag for the developer so he/she doesn't just blindly commit them...